packages/e2e-test-utils-playwright/src/request-utils/global-styles.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Internal dependencies
+ */
+import type { RequestUtils } from './index';
+
+/**
+ * Get the current user global styles post id.
+ *
+ * @see https://developer.wordpress.org/rest-api/reference/themes/#definition
+ * @param {RequestUtils} this RequestUtils.
+ * @return {Promise<string>} The user global styles post id.
+ */
+async function getCurrentThemeGlobalStylesPostId( this: RequestUtils ) {
+	type ThemeItem = {
+		stylesheet: string;
+		status: string;
+		_links: { 'wp:user-global-styles': { href: string }[] };
+	};
+	const themes = await this.rest< ThemeItem[] >( {
+		path: '/wp/v2/themes',
+	} );
+	let themeGlobalStylesId: string = '';
+	if ( themes && themes.length ) {
+		const currentTheme: ThemeItem | undefined = themes.find(
+			( { status } ) => status === 'active'
+		);
+
+		const globalStylesURL =
+			currentTheme?._links?.[ 'wp:user-global-styles' ]?.[ 0 ]?.href;
+		if ( globalStylesURL ) {
+			themeGlobalStylesId = globalStylesURL?.split(
+				'rest_route=/wp/v2/global-styles/'
+			)[ 1 ];
+		}
+	}
+	return themeGlobalStylesId;
+}
+
+/**
+ * Update the user's global styles. Note that this overwrites and previously
+ * set styles. Styles that you want to keep should be merged in manually.
+ *
+ * @param {RequestUtils} this   RequestUtils.
+ * @param {Object}       styles A theme.json compatible object.
+ */
+async function updateGlobalStyles( this: RequestUtils, styles: Object ) {
+	const globalStylesPostId = await this.getCurrentThemeGlobalStylesPostId();
+
+	if ( ! globalStylesPostId ) {
+		return;
+	}
+
+	return await this.rest( {
+		path: `/wp/v2/global-styles/${ globalStylesPostId }`,
+		method: 'POST',
+		data: { id: globalStylesPostId, styles },
+	} );
+}
+
+/**
+ * List all post revisions using the REST API.
+ *
+ * @param {RequestUtils}  this     RequestUtils.
+ * @param {string|number} parentId Post attributes.
+ */
+async function getThemeGlobalStylesRevisions(
+	this: RequestUtils,
+	parentId: number | string
+) {
+	// Lists all global styles revisions.
+	return await this.rest< Record< string, Object >[] >( {
+		path: `/wp/v2/global-styles/${ parentId }/revisions`,
+		params: {
+			per_page: 100,
+		},
+	} );
+}
+
+export {
+	getCurrentThemeGlobalStylesPostId,
+	getThemeGlobalStylesRevisions,
+	updateGlobalStyles,
```

Comment:
@talldan, do you think these utils will be helpful in other tests? The best practice is to avoid overloading global utils in favor of local utils - https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/README.md#dont-overload-test-utils-inline-simple-utils.

Reply:
Two of them were already present, I just moved them to a different file.

`updateGlobalStyles` is the only new one, which I think will be very useful, as I plan to add some more of these tests for different blocks and combinations of blocks.

It might also be useful for clearing global styles changes as part of the test cleanup, though perhaps we also want to clear revisions when doing that.

Reply:
`updateGlobalStyles` is potentially helpful for testing revisions, see: https://github.com/WordPress/gutenberg/blob/add/e2e-tests-for-block-styles/test/e2e/specs/site-editor/user-global-styles-revisions.spec.js#L304 👍🏻